src/main/java/seedu/address/logic/parser/DeleteStudentCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENTID;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByEmailCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByIdCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByIndexCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.StudentId;
+
+/**
+ * Parses input arguments and creates a new DeleteCommand object
+ */
+public class DeleteStudentCommandParser implements Parser<DeleteStudentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the
+     * DeleteCommand
+     * and returns a DeleteCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteStudentCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_INDEX, PREFIX_EMAIL, PREFIX_STUDENTID);
+        boolean isIndexPresent = argMultimap.getValue(PREFIX_INDEX).isPresent();
+        boolean isEmailPresent = argMultimap.getValue(PREFIX_EMAIL).isPresent();
+        boolean isStudentIdPresent = argMultimap.getValue(PREFIX_STUDENTID).isPresent();
+        if (!isEmailPresent && !isStudentIdPresent && !isIndexPresent
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteStudentCommand.MESSAGE_USAGE));
+        }
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_INDEX, PREFIX_STUDENTID, PREFIX_EMAIL);
+        if (isIndexPresent) {
+            Index index = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+            return new DeleteStudentByIndexCommand(index);
+        } else if (isEmailPresent) {
+            Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+            return new DeleteStudentByEmailCommand(email);
+        } else if (isStudentIdPresent) {
+            StudentId studentId = ParserUtil.parseStudentId(argMultimap.getValue(PREFIX_STUDENTID).get());
+            return new DeleteStudentByIdCommand(studentId);
+        } else {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
```

Comment:
Does this mean that specifying index to delete is prioritised over email and studentid? This can happen when the user specifies all the fields right? for example "delete email/john@gmail.com id/A1231231L index/1

Reply:
Yes. The priority is as follows: Index, Student Id, Email.